frontend/src/components/activePage/Instances/InstanceIcons/InstanceIcons.tsx

```diff
@@ -0,0 +1,53 @@
+import { FC } from 'react';
+import { Avatar } from 'antd';
+import {
+  CodeOutlined,
+  DesktopOutlined,
+  CheckCircleOutlined,
+  LoadingOutlined,
+  CloseCircleOutlined,
+  PoweroffOutlined,
+} from '@ant-design/icons';
+
+export interface IInstanceIconsProps {
+  isGUI: boolean;
+  phase: 'ready' | 'creating' | 'failed' | 'stopping' | 'off';
+}
+
+const InstanceIcons: FC<IInstanceIconsProps> = ({ ...props }) => {
+  const { isGUI, phase } = props;
+
+  const chooseStatusIcon = () => {
```

Comment:
Just a suggestion, you could define a object (outside the component) with:
- key: phase
- value: component corresponding to the phase
And then instead of calling the function you take the value from the object